### ISSUE TITLE
Can't install Windows 8.1 SDK (Microsoft.VisualStudio.Component.Windows81SDK)

### ISSUE THREAD

**Issue by nsleigh** [jberezanski/ChocolateyPackages#27]
I am trying to add the Windows 8.1 SDK to the vctools install using the following command:

`choco install visualstudio2017-workload-vctools --package-parameters "--add Microsoft.VisualStudio.Component.Windows81SDK"`

No matter what combinations of includeRecommended and includeOptional I use it fails to install. If I run the command manually (e.g. launching the vctools.exe directly it is added). Also, if I use the same command with the visualstudio2017buildtools package it works.

Sound this command work?

**Comment by jberezanski** [comment]
Are you trying to add the vctools workload, together with the Windows81SDK optional component, to an instance of Build Tools which does not have the vctools workload yet? Or are you trying to add Windows81SDK to an instance of Build Tools which already has that workload installed?

Also, I'd like to look at your chocolatey.log (C:\ProgramData\Chocolatey\Logs\chocolatey.log), it will help me understand the state of your machine. You can, for example, upload it as a [gist](https://gist.github.com).

**Comment by nsleigh** [comment]
The first one: _add the vctools workload, together with the Windows81SDK optional component, to an instance of Build Tools which does not have the vctools workload yet._

I will recreate the environment and get the log file.

**Comment by jberezanski** [comment]
Thanks! I believe it should work, so if it doesn't, I'd call it a bug.

**Comment by aallrd** [comment]
Hello,
I am facing the exact same issue when trying to install the **Microsoft.VisualStudio.Component.Windows81SDK** component on the **visualstudio2017-workload-nativedesktop** workload.
I am running these commands in a container:
```
RUN choco install visualstudio2017enterprise; \
        choco install visualstudio2017-workload-nativedesktop; \
        choco install visualstudio2017-workload-nativedesktop --package-parameters "--add Microsoft.VisualStudio.Component.Windows81SDK"
```

I get the below error:
```
Chocolatey v0.10.8
Installing the following packages:
visualstudio2017-workload-nativedesktop;Microsoft.VisualStudio.Component.Windows81SDK
By installing you accept licenses for the packages.
visualstudio2017-workload-nativedesktop v1.1.1 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
Microsoft.VisualStudio.Component.Windows81SDK not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"

Chocolatey installed 0/2 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Warnings:
 - visualstudio2017-workload-nativedesktop - visualstudio2017-workload-nativedesktop v1.1.1 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.

Failures
 - Microsoft.VisualStudio.Component.Windows81SDK - Microsoft.VisualStudio.Component.Windows81SDK not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"
The command 'powershell.exe -ExecutionPolicy Bypass -Command choco install visualstudio2017-workload-nativedesktop --package-parameters "--add Microsoft.VisualStudio.Component.Windows81SDK"' returned a non-zero code: 1
```

I also created a gist with the corresponding chocolatey.log: [Chocolatey failed Microsoft.VisualStudio.Component.Windows81SDK installation](https://gist.github.com/aallrd/a6b7f8d38606564851d70d6f7a1462f9#file-chocolatey-failed-microsoft-visualstudio-component-windows81sdk-installation)

Is the command I am using the correct one?

**Comment by aallrd** [comment]
For reference,  I was able to add the Microsoft.VisualStudio.Component.Windows81SDK component like this:
```
RUN & 'C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installer.exe' \
	modify --quiet --norestart --nocache \
	--installPath 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise' \
	--add Microsoft.VisualStudio.Component.Windows81SDK; \
	Write-Host 'Waiting for vs_installer.exe to finish...'; \
	Wait-Process -Name vs_installer
```

**Comment by jberezanski** [comment]
@aallrd 

> RUN choco install visualstudio2017enterprise; \
        choco install visualstudio2017-workload-nativedesktop; \
        choco install visualstudio2017-workload-nativedesktop --package-parameters "--add Microsoft.VisualStudio.Component.Windows81SDK"

> I get the below error:

> Chocolatey v0.10.8
> Installing the following packages:
> visualstudio2017-workload-nativedesktop;Microsoft.VisualStudio.Component.Windows81SDK

Definitely a quoting issue between docker and choco.exe, not a package problem. The quotes do not get passed to choco.exe and it thinks Microsoft.VisualStudio.Component.Windows81SDK is another package id to install.

**Comment by jberezanski** [comment]
@nsleigh It turns out there was a flaw in the chocolatey-visualstudio.extension (a module used by the workload packages) that caused --add to be ignored for workload packages (it worked for product packages such as visualstudio2017buildtools, but only a single --add could be added). Both flaws [have been corrected](https://github.com/jberezanski/ChocolateyPackages/blob/master/chocolatey-visualstudio.extension/CHANGELOG.md) in the newest release, which is available right now as a preview ([version 1.6.0-rc7](https://chocolatey.org/packages/chocolatey-visualstudio.extension/1.6.0-rc7)) and will be made official probably in the next couple of days.

Example command lines:
````
# update the extension to the prerelease version
cup chocolatey-visualstudio.extension --pre

# install VS 2017 Professional with two optional components and an additional language
cinst -y visualstudio2017professional --params "--add Microsoft.Net.Component.4.7.TargetingPack --add Microsoft.Net.Component.4.7.1.TargetingPack --locale en-US --addProductLang de-DE"

# add the Data workload and two additional components to all installed VS 2017 instances, skipping components recommended for that workload (they would be installed by default)
cinst -y visualstudio2017-workload-data --params "--add Microsoft.VisualStudio.Component.WebDeploy --add Microsoft.VisualStudio.Component.VisualStudioData --locale en-US --no-includeRecommended"
````

**Comment by jberezanski** [comment]
The fixed extension is now [official](https://chocolatey.org/packages/chocolatey-visualstudio.extension/1.6.0).
